urlx.go

```diff
@@ -13,23 +13,44 @@ import (
 	"golang.org/x/net/idna"
 )
 
-// Parse parses raw URL string into the net/url URL struct.
-// It uses the url.Parse() internally, but it slightly changes
-// its behavior:
-// 1. It forces the default scheme and port.
-// 2. It favors absolute paths over relative ones, thus "example.com"
-//    is parsed into url.Host instead of url.Path.
-// 4. It lowercases the Host (not only the Scheme).
-func Parse(rawURL string) (*url.URL, error) {
+type Transformer func(rawURL string) string
```

Comment:
Hi, thank you for your contribution! I think having the default scheme is the right call, but I'm not convinced we need Transformer functions.

Can you think of any other use cases?

Reply:
The other use cases that come to mind are mainly focused around defaults, though I can imagine some other uses cases but none of them are ones that I have nor need at this time.

I decided to go the generic transform route because I didn't want have a very specific `ParseWithDefaultScheme` function, especially if additional defaults were to be added in the future. My current use case is to have the default `https` scheme, but nothing beyond that. My thought was to have a generic Transform function that way you could add additional transforms as they are needed without having to change the function signature.

Reply:
Got it. I don't have any use cases for Transformer either.

How about we scratch that and have a simple `SetDefaultScheme(scheme string)` function instead? The default value (private variable) would be https.

Reply:
The problem with that is that it would end up being a global variable. What if I want to parse some URLs with a default scheme of `https` and some with a default of `ftp` or `http` or any other scheme? I'd have to reset the global default scheme before every parse rather than it having a limited scope.

Reply:
Good point.

Imho all we need is `ParseWithDefaultScheme(url string, scheme string) string`, if I'm not wrong.

Reply:
Updated. Please review. Thanks!